OPHD/States/MapViewStateTurn.cpp

```diff
@@ -89,15 +89,10 @@ void MapViewState::updatePopulation()
 	int nurseries = structureManager.getCountInState(Structure::StructureClass::Nursery, StructureState::Operational);
 	int hospitals = structureManager.getCountInState(Structure::StructureClass::MedicalCenter, StructureState::Operational);
 
-	auto foodproducers = structureManager.structureList(Structure::StructureClass::FoodProduction);
-	auto& command = structureManager.structureList(Structure::StructureClass::Command);
-	foodproducers.insert(foodproducers.end(), command.begin(), command.end());
-
 	int remainder = mPopulation.update(mCurrentMorale, mFood, residences, universities, nurseries, hospitals);
 
-	for (auto structure : foodproducers)
+	for (auto& foodProducer : enumerateComponent<FoodProduction>())
```

Comment:
I ditched the "consume from Command Centers last" behavior as it didn't fit in well with the notion that all food producing components are the same. In practice the difference will be minor since `MapViewState::transferFoodToCommandCenter` will make sure the CCs are near-full anyway (full minus up to one turn's consumption).

Reply:
Fair enough. I find this acceptable.